bin/i18n/sync-out.rb

```diff
@@ -258,6 +258,21 @@ def sanitize_data_and_write(data, dest_path)
   File.write(dest_path, dest_data)
 end
 
+def sort_and_sanitize(hash)
+  hash.sort_by {|key, _| key}.each_with_object({}) do |(key, value), result|
+    if value.is_a? Hash
+      sorted_value = sort_and_sanitize(value)
+      result[key] = sorted_value unless sorted_value.nil? || sorted_value.empty?
+    elsif value.is_a? Array
+      result[key] = value.filter_map {|v| v.is_a?(Hash) ? sort_and_sanitize(v) : v}
+    elsif value.is_a? String
+      result[key] = value.gsub(/\\r/, "\r")
+    else
+      result[key] = value unless value.nil?
+    end
```

Comment:
```suggestion
    case value
    when Hash
      sorted_value = sort_and_sanitize(value)
      result[key] = sorted_value unless sorted_value.nil? || sorted_value.empty?
    when Array
      result[key] = value.filter_map {|v| v.is_a?(Hash) ? sort_and_sanitize(v) : v}
    when String
      result[key] = value.gsub(/\\r/, "\r")
    else
      result[key] = value unless value.nil?
    end
```

Reply:
Good catch, the case statement will definitely clean this up. Is there Ruby magic that allows me to do the `when SomeClass` statements or do I need to do `case value.class`?

Reply:
You can simply use `value` without `#class` in case statements